src/agentscope/parsers/json_object_parser.py

```diff
@@ -139,11 +141,22 @@ class MarkdownJsonDictParser(MarkdownJsonObjectParser, DictFilterMixin):
     """Closing end for a code block."""
```

Comment:
A suggestion, maybe we can re-implement `DictFilterMixin.to_metadata` to assign `x.metadata` to `BaseModel` type which would help developers can reference `x.metadata.some_property` directly

```python3
from typing import TypeVar, Type
from pydantic import Field, BaseModel


GenericResponseModel = TypeVar('GenericResponseModel', bound=BaseModel)


class MarkdownJsonDictParser(MarkdownJsonObjectParser, DictFilterMixin):
    def to_metadata(
        self,
        parsed_response: dict,
        allow_missing: bool = False,
    ) -> Union[str, dict, GenericResponseModel, None]:
        """Filter the fields that will be fed into the returned message
        directly to control the application workflow."""
        return self._filter_content_by_names(
            parsed_response,
            self.keys_to_metadata,
            allow_missing=allow_missing,
        )
```

And here is a code snippet for quick test
```python3
from typing import TypeVar, Type, Union
from pydantic import Field, BaseModel


GenericResponseModel = TypeVar('GenericResponseModel', bound=BaseModel)


class EndDiscuss(BaseModel):
    thought: str = Field(description="what you thought")
    speak: str = Field(description="what you speak")
    end_discussion: bool = Field(description="whether the discussion ends")


class VotePlayer(BaseModel):
    thought: str = Field(description="what you thought")
    speak: str = Field(description="what you speak")
    whether_vote: bool = Field(description="whether vote some other player")


def generate_example_response(model: Type[GenericResponseModel]) -> Union[str, dict, GenericResponseModel, None]:
    return model(**{field_name: field_info.annotation() for field_name, field_info in model.model_fields.items()})


ed = generate_example_response(EndDiscuss)
vp = generate_example_response(VotePlayer)

print(ed.end_discussion, vp.whether_vote)
```

By the way, I've considered to change `ModelResponse.__call__` return type to `GenericResponseModel`. But I'm afraid it is a too breaking change that exceed the scope of this PR

Reply:
We have opened discussion panel in this repo! Welcome to have a discussion there if needed.